lib/spack/spack/database.py

```diff
@@ -698,7 +706,7 @@ def query_one(self, query_spec, known=any, installed=True):
         query. Returns None if no installed package matches.
 
         """
-        concrete_specs = self.query(query_spec, known, installed)
+        concrete_specs = self.query(query_spec, known, installed, unique=True)
         assert len(concrete_specs) <= 1
```

Comment:
Do you agree that we will never hit this line:
```python
assert len(concrete_specs) <= 1
```
because if `unique=True` and `len(results) == 0` the program will `tty.die`?

I think we shouldn't change anything in the DB API, but rather act in `cmd/module.py`